### DISCUSSION TITLE
Remove common module from child compilation

### DISCUSSION THREAD

**Discussion by samagragupta** [webpack/webpack#16406] [Q&A]
I am running a child compilation whose entry is not same as the parent compilation. The entry for child compilation is some another file (Hello.js, not index.js).

Now I want the parent compiler to skip the compilation from Hello.js. So that no extra modules are added in the parent chunks. As in the end I will bundle both these chunks. I want to skip it like that after bundling both chunks from parent and child , it should work perfectly fine.

```
class AwesomePlugin {
  constructor() {
    this.childCompilerName = 'awesome-plugin-compilation';
    this.outputFileName = 'custom-file.js';
    this.compilationEntry = './src/Hello.js';
  }
  apply(compiler) {
    compiler.hooks.emit.tapAsync('AwesomePlugin', (compilation, callback) => {
      const childCompiler = compilation.createChildCompiler(this.childCompilerName, {
        filename: this.outputFileName
      });
      childCompiler.options.output.chunkFilename = `child-${compiler.options.output.chunkFilename}`;
      childCompiler.context = compiler.context;
      childCompiler.apply(new SingleEntryPlugin(compiler.context, this.compilationEntry, this.outputFileName));
      childCompiler.plugin('compilation', (compilation) => {
        if (compilation.cache) {
        }
      });
      childCompiler.runAsChild((err, entries, childCompilation) => {
        callback(err);
      });
    });
  }
}
```

**Comment by alexander-akait** [comment]
You can pass aliases or externals to child compilation and replace some modules on empty modules (which will be removed), if you want to find intersec between parent and child compilation you need travers graph, it will be more slow, what is use case?

**Reply by samagragupta** [reply]
I want to create two different chunks. And at request time, I will bundle both these chunks and serve it.
For example we have file World.js(which is imported dynamically/loadable). World.js is importing two other files that is Hello.js and Bye.js.
Now currently there will be one chunk (World.chunk.js) which will have World.js, Hello.js and Bye.js.

But now I will traverse through AST, find special identifier(most probably a comment) for Hello.js and Bye.js. After that I want to create 3 files(or chunks) of Hello.chunk.js, Bye.chunk.js and World.chunk.js(does not contain Hello.js and Bye.js)
Now during request time, I wil bundle World.chunk.js and Hello.chunk.js or some other combination.

**Reply by samagragupta** [reply]
@alexander-akait Or if we can skip parent compilation for which child compilation is working?

**Reply by alexander-akait** [reply]
@samagragupta To be honetly I don't undestand why you need this (because it looks strange idea), but I think you don't need parent compilation here, you can use multiple chunks or multi compilations mode

**Reply by samagragupta** [reply]
@alexander-akait But I don't think through this I can achieve that because these configuration should be given at the start of compilation. But as I said I want to make a new file for that module which I get to know after traversing through AST and finding special identifier for that module.

**Reply by alexander-akait** [reply]
Why you really need this, Sounds like you try to achive something exotic, that is why I think you need to change approach to get the result

**Reply by samagragupta** [reply]
@alexander-akait I want this so that I can solve AB problem. For a user, AB specific code to be served. No extra code (other AB). So I want to to create different files (chunks) for those AB files and bundle them during runtime.
Ref: https://www.youtube.com/watch?v=aB5i4JhgwB8

**Reply by alexander-akait** [reply]
You can just using `import(...)` to loader condition based changes or you can provide options/configuration on `window` level, I don't think you need to bundle, because you will have extra code (webpack runtime), also you can use `layer` to bundle one chunk multiple times with different options
